src/hooks/useSwapCallbackV3.ts

```diff
@@ -0,0 +1,140 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { TransactionResponse } from '@ethersproject/providers'
+import { Currency, CurrencyAmount } from '@kyberswap/ks-sdk-core'
+import { useCallback } from 'react'
+
+import { ZERO_ADDRESS_SOLANA } from 'constants/index'
+import { useActiveWeb3React, useWeb3React } from 'hooks/index'
+import useENS from 'hooks/useENS'
+import { useSwapState } from 'state/swap/hooks'
+import { useTransactionAdder } from 'state/transactions/hooks'
+import { TRANSACTION_TYPE } from 'state/transactions/type'
+import { useUserSlippageTolerance } from 'state/user/hooks'
+import { isAddress, shortenAddress } from 'utils'
+import { formatCurrencyAmount } from 'utils/formatCurrencyAmount'
+import { sendEVMTransaction } from 'utils/sendTransaction'
+
+export interface FeeConfig {
+  chargeFeeBy: 'currency_in' | 'currency_out'
+  feeReceiver: string
+  isInBps: boolean
+  feeAmount: string
+}
+
+// returns a function that will execute a swap, if the parameters are all valid
+// and the user has approved the slippage adjusted input amount for the trade
+const useSwapCallbackV3 = (
```

Comment:
can we adjust useSwapV2CallBack to reuse for new swap form?

Reply:
I'm afraid not. `useSwapCallbackV3` uses SwapForm context